rts/Lua/LuaImage.cpp

```diff
@@ -0,0 +1,412 @@
+}
+
+
+std::shared_ptr<LuaImageData> LuaImage::LoadImageObject(lua_State* L)
+{
+	std::string filename = luaL_checkstring(L, 1);
+	int channels = LuaImage::ParseFormat(L, 2);
+	int dataType = LuaImage::ParseDataType(L, 3);
+	bool luminance = luaL_optboolean(L, 4, false);
+	auto image = make_shared<LuaImageData>(filename, channels, dataType, luminance);
+	if (image->valid) {
+		return image;
+	}
+	return nullptr;
+}
+
+
+/******************************************************************************/
+/******************************************************************************/
+
+int LuaImage::meta_gc(lua_State* L)
+{
+	if (lua_isnil(L, 1))
+		return 0;
+
+	auto image = std::move(*static_cast<std::shared_ptr<LuaImageData>*>(luaL_checkudata(L, 1, "Image")));
+	image = {};
+
+	return 0;
+}
+
+
+int LuaImage::meta_tostring(lua_State* L)
+{
+	if (lua_isnil(L, 1))
+		return 0;
+
+	auto image = toimage(L, 1);
+	lua_pushstring(L, std::format("Image(\"{}\", 0x{:x})", image->filename, reinterpret_cast<intptr_t>(image)).c_str());
+
+	return 1;
+}
+
+
+int LuaImage::meta_index(lua_State* L)
+{
+	// first check if there is a function
+	luaL_getmetatable(L, "Image");
+	lua_pushvalue(L, 2);
+	lua_rawget(L, -2);
+	if (!lua_isnil(L, -1))
+		return 1;
+
+	lua_pop(L, 1);
+
+	// couldn't find a function, so check properties
+	auto image = toimage(L, 1);
+
+	if (lua_israwstring(L, 2)) {
+		switch (hashString(lua_tostring(L, 2))) {
+			case hashString("width"): {
+				lua_pushinteger(L, image->width);
+				return 1;
+			} break;
+
+			case hashString("height"): {
+				lua_pushinteger(L, image->height);
+				return 1;
+			} break;
+
+			case hashString("size"): {
+				lua_pushinteger(L, image->width);
+				lua_pushinteger(L, image->height);
+				return 2;
+			} break;
+
+			case hashString("channels"): {
+				lua_pushinteger(L, image->channels);
+				return 1;
+			} break;
+
+			case hashString("format"): {
+				lua_pushstring(L, ChannelsToFormat(image->channels));
+				return 1;
+			} break;
+
+			case hashString("dataType"): {
+				lua_pushstring(L, DataTypeToString(image->bitmap->dataType));
+				return 1;
+			} break;
+
+			default: {
+			} break;
+		}
+	}
+
+	return 0;
+}
+
+
+/******************************************************************************
+ *  Api
+ */
+
+/*** Loads an image
+ *
+ * @function Image.LoadImage
+ *
+ * @param filePath string VFS path to the file, for example "map/windfield.png".
+ * @param format string Force resulting Image format. Leave as "default" or nil for default.
+ * @param dataType string Force resulting Image dataType. Leave as "default" or nil for default.
+ *
+ * @return Image userdata object to operate on the image
+ */
+
+int LuaImage::LoadImage(lua_State* L)
+{
+	RECOIL_DETAILED_TRACY_ZONE;
+	auto image = LoadImageObject(L);
+	if (image == nullptr)
+		return 0;
+
+	auto shPtrImagePtr = static_cast<decltype(image)*>(lua_newuserdata(L, sizeof(decltype(image))));
+	memset(shPtrImagePtr, 0, sizeof(decltype(image)));
```

Comment:
Operator `sizeof` works on values too, not just types 
```suggestion
	memset(shPtrImagePtr, 0, sizeof image);
```

Reply:
could have suggested the line above too :D